src/web/mjs/connectors/VizShonenJump.mjs

```diff
@@ -34,26 +36,86 @@ export default class VizShonenJump extends Connector {
         };
     }
 
-    async _getMangas() {
+    async _getMangasAvailibleByVolumes() {
+        const request = new Request(new URL('/library', this.url), this.requestOptions);
+        let data = await this.fetchDOM(request);
+
+        data = [...data.querySelectorAll('table.purchase-table a')];
+
+        return data.map(manga => {
+            return {
+                volume_id: manga.pathname,
+                title: manga.innerText.trim(),
+                entry_type: this.EntryTypeEnum.volume
+            };
+        });
+    }
+
+    async _getMangasAvailibleByChapters() {
         const request = new Request(new URL('/shonenjump', this.url), this.requestOptions);
         let data = await this.fetchDOM(request);
 
         if ( !data.innerText.includes('Latest free chapters') ) {
             alert('This website is geolocked. It can only be accessed from the USA.\nYou may use MANGA Plus instead.', this.label, 'info');
             return;
         }
-
-        data = [...data.querySelectorAll('div.o_sort_container div > a')];
+        
+        data = [...data.querySelectorAll('a.o_chapters-link')];
 
         return data.map(manga => {
             return {
-                id: manga.pathname,
-                title: manga.querySelector(':nth-child(2)').innerText.trim()
+                id: JSON.stringify({
+                    id: manga.pathname,
+                    title: manga.innerText.trim(),
+                    entry_type: this.EntryTypeEnum.chapter
+                }),
+                title: manga.innerText.trim(),
             };
         });
     }
 
-    async _getChapters(manga) {
+    async _getMangas() {
+        const mangaAvailibleByVolumes = await this._getMangasAvailibleByVolumes();
+        let mangaAvailibleByChapters = await this._getMangasAvailibleByChapters();
+
+        mangaAvailibleByVolumes.forEach(mangaAvailibleByVolume => {
+                const element = mangaAvailibleByChapters.find(element => element.title === mangaAvailibleByVolume.title);
+                // Found it.
+                if (element != undefined) {
+                    let complexManga = JSON.parse(element.id);
+
+                    element = {
+                        id: JSON.stringify({
+                            id: complexManga.id,
+                            volume_id: mangaAvailibleByVolume.volume_id,
+                            title: mangaAvailibleByVolume.title,
+                            entry_type: this.EntryTypeEnum.volume | this.EntryTypeEnum.chapter
```

Comment:
why | and not + ?
(this is what threw me on the bitwize comment)
`this.EntryTypeEnum.volume + this.EntryTypeEnum.chapter`